pkg/shared/service.go

```diff
@@ -17,7 +19,139 @@ func NewServiceHandler() *ServiceHandler {
 }
 
 func (sh *ServiceHandler) Handle(ctx context.Context, event sdk.Event) error {
-	fmt.Println("handling object ", event.Object.GetObjectKind().GroupVersionKind().String())
+	logrus.Debug("handling object ", event.Object.GetObjectKind().GroupVersionKind().String())
+
+	sharedService := event.Object.(*v1alpha1.SharedService)
+	sharedServiceCopy := sharedService.DeepCopy()
+
+	if sharedService.Spec.ServiceType != strings.ToLower(v1alpha1.KeycloakKind) {
+		return nil
+	}
+
+	if event.Deleted {
+		return nil
+	}
+
+	if sharedService.GetDeletionTimestamp() != nil {
+		return sh.finalizeSharedService(sharedServiceCopy)
+	}
+
+	logrus.Debugf("SharedServicePhase: %v", sharedService.Status.Phase)
+	switch sharedService.Status.Phase {
+	case v1alpha1.SSPhaseNone:
+		sh.initSharedService(sharedServiceCopy)
+	case v1alpha1.SSPhaseAccepted:
+		sh.createKeycloaks(sharedServiceCopy)
+	}
+
+	return nil
+}
+
+func (sh *ServiceHandler) createKeycloaks(sharedService *v1alpha1.SharedService) error {
+	keycloakList := v1alpha1.KeycloakList{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Keycloak",
+			APIVersion: "aerogear.org/v1alpha1",
+		},
+	}
+	listOptions := sdk.WithListOptions(&metav1.ListOptions{
+		LabelSelector:        "aerogear.org/sharedServiceName=" + sharedService.ObjectMeta.Name,
+		IncludeUninitialized: false,
+	})
+
+	err := sdk.List(sharedService.Namespace, &keycloakList, listOptions)
+	if err != nil {
+		logrus.Errorf("Failed to query keycloaks : %v", err)
+		return err
+	}
+
+	numCurrentInstances := len(keycloakList.Items)
+	numRequiredInstances := minRequiredInstances(sharedService.Spec.MinInstances, sharedService.Spec.RequiredInstances)
+	numMaxInstances := sharedService.Spec.MaxInstances
```

Comment:
This isn't currently checking max instances, but i'm assuming we will in following tickets when we deal with scaling.

Reply:
yep, that's fine